src/reformat/rgb_impl.rs

```diff
@@ -258,19 +266,15 @@ pub fn yuv_to_rgb_any(
         && image.yuv_format != PixelFormat::Monochrome;
     let yuv_max_channel = state.yuv.max_channel;
     let rgb_max_channel_f = state.rgb.max_channel_f;
-    for j in 0..image.height as usize {
+    for j in 0..image.height {
         let uv_j = j >> image.yuv_format.chroma_shift_y();
-        let y_row = image.row(Plane::Y, j as u32);
-        let u_row = image.row(Plane::U, uv_j as u32);
-        let v_row = image.row(Plane::V, uv_j as u32);
-        let a_row = image.row(Plane::A, uv_j as u32);
-        let y_row16 = image.row16(Plane::Y, j as u32);
-        let u_row16 = image.row16(Plane::U, uv_j as u32);
-        let v_row16 = image.row16(Plane::V, uv_j as u32);
-        let a_row16 = image.row16(Plane::A, j as u32);
-        let (mut rgb_row, mut rgb_row16) = rgb.rows_mut(j as u32)?;
+        let y_row = image.row_generic(Plane::Y, j);
```

Comment:
if we make row_generic return Optional. You can unwrap it here with `ok_or(...)?;`

that way we don't have to deal with the row being `Absent` in the helper functions.

Reply:
yeah even the `u_row` and `v_row` can be `None` in case of monochrome images. so i think it is better to just unwrap them in `clamped_pixel()` as you are doing now.

i was just wondering if we could do the unwrapping at a row level instead of at a pixel level.